mage/query-modules/python/gnn-link-prediction.md

```diff
@@ -0,0 +1,375 @@
+- `validation results: List[Dict[str, float]]` -> List of validation results through epochs. Model's performance is evaluated every `console_log_freq` epochs. 
+
+### **Predict**
+
+Predict method takes two arguments, **src_vertex** and **dest_vertex**, and predicts whether there is an edge between them or not. It supports an `“actual”` prediction scenario when the edge doesn’t exist and the user wants to predict whether there is an edge or not but also a scenario in which there is an edge between two vertices and the user wants to check the model’s evaluation. 
+
+#### Input
+- `src_vertex: mgp.Vertex` -> Source vertex of the edge
+- `dest_vertex: mgp.Vertex` -> Destination vertex of the edge.
+
+#### Output
+- `score: mgp.Number` -> Score between 0 and 1 that represents the probability of two nodes being connected. 
+
+```
+MATCH (v1:PAPER {id: "ID_1"})
+MATCH (v2:PAPER {id: "ID_2"})
+CALL link_prediction.predict(v1, v2)
+YIELD score
+RETURN score;
+```
+
+### **Recommend**
+
+Recommend method can be used to recommend the best k nodes from dest_vertices to src_vertex. It is implemented efficiently using the **max heap** data structure. The best nodes are determined based on the edge scores. Metrics specific to recommendation systems (**precision@k, recall@k, f1@k and average precision**) are logged to the **standard output**. **K** is equal to the given `min(k, length(dest_vertices), length(results))` where results are a list of all recommendations given by the model(classified as a positive example.)
+
+#### Input
+- `src_vertex: mgp.Vertex` → Source node.
+- `dest_vertices: List[mgp.Vertex]` → destination nodes. If they are not of the same type, an exception is thrown.
+- `k: int` → Number of edges to recommend. 
+
+#### Output
+- `score: mgp.Number` → Score between 0 and 1 that represents the probability of two nodes being connected.
+- `recommendation: mgp.Vertex` → A reference to the target node.
+
+```
+MATCH (v1:Customer {id: "8779-QRDMV"})
+MATCH (p:Plan) 
+WITH collect(p) AS all_plans, v1 
+CALL link_prediction.recommend(v1, all_plans, 5)
+YIELD score, recommendation
+RETURN v1, score, recommendation;
+```
+
+### **Load context**
+
+Loading the context means loading the model and the predictor. If the user specifies the path, the method will try to load it from there. Otherwise, context will be loaded from the default parameter specified in the link_prediction_parameters module.
+
+#### Input
+- `path: str` → Path to the folder where the model and the predictor are saved.
+
+#### Output
+- `status: mgp.Any` → True to indicate that execution went well.
+
+```
+CALL link_prediction.load_context() YIELD * RETURN *;
+```
+
+### **Reset parameters**
+
+User can explicitly reset parameters whenever he/she wants. Note, however, that parameters will be reset before the training even if not specified by the user because of implementation reasons.
```

Comment:
```suggestion
You can explicitly reset parameters whenever you want. Note, however, that parameters will be reset before the training even if not specified because of implementation reasons.
```

Or decide whether you want to use USER through the whole docs and write general rules or YOU and give instructions